### PR TITLE
refactor(aria-hidden-body): work with virtual nodes

### PR DESCRIPTION
**Pull request by shankarshastri**
Closes issue: #2475 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [X] Follows the commit message policy, appropriate for next version
- [X] Code is reviewed for security

### CODE REVIEW

**Comment by shankarshastri** [comment]
@straker , I have added the draft PR for the same, even after adding the id=target, its having issues. Could you please let me know how I can proceed?

**Comment by straker** [comment]
Ohhhh. Sorry, I see the problem now. `queryFixture` tries to append the HTML content to the body, but that isn't allowed as you cant add `body` to itself. So nothing is being added and thus nothing returned. We'll have to go around that and setup the virtual tree for every test and get the body element to pass to it.

```js
it('should not be present on document.body', function() {
  var tree = axe.testUtils.flatTreeSetup(document.body);
  assert.isTrue(
    axe.testUtils
      .getCheckEvaluate('aria-hidden-body')
      .call(checkContext, null, {}, tree[0])
  );
});
```

File: lib/checks/aria/aria-hidden-body-evaluate.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
-function ariaHiddenBodyEvaluate(node) {
-	return node.getAttribute('aria-hidden') !== 'true';
+function ariaHiddenBodyEvaluate(node, options, virtualNode) {
+	return (
+		virtualNode.hasAttr('aria-hidden') !== true ||
+		virtualNode.attr('aria-hidden') === 'false'
+	);
```
Comment (straker) [inline lib/checks/aria/aria-hidden-body-evaluate.js:5]: Interesting change. Did this solve a problem you ran into? Was `virtualNode.attr('aria-hidden') !== 'true'` not working?

Comment (shankarshastri) [reply lib/checks/aria/aria-hidden-body-evaluate.js:5]: My bad, fixed it. I was actually exploring the virtual-node js file and tried multiple functions of the same.

**Pull request merged**